docs/bigdata/README.md

```diff
@@ -0,0 +1,272 @@
+
+## **2. Prerequisites**
+
+*   Install Hortonworks Distribution using this [guide.](https://docs.hortonworks.com/HDPDocuments/Ambari-2.7.1.0/bk_ambari-installation/content/ch_Installing_Ambari.html)
+    *   [Setup Ambari](https://docs.hortonworks.com/HDPDocuments/Ambari-2.7.1.0/bk_ambari-installation/content/set_up_the_ambari_server.html) which automatically sets up YARN
+    *   [Installing Spark](https://docs.hortonworks.com/HDPDocuments/HDP3/HDP-3.0.1/installing-spark/content/installing_spark.html)
+*   Install MinIO Distributed Server using one of the guides below.
+    *   [Deployment based on Kubernetes](https://docs.min.io/docs/deploy-minio-on-kubernetes.html#minio-distributed-server-deployment)
+    *   [Deployment based on MinIO Helm Chart](https://github.com/helm/charts/tree/master/stable/minio)
+
+## **3. Configure Hadoop, Spark, Hive to use MinIO**
+
+After successful installation navigate to the Ambari UI http://<ambari-server>:8080/ and login using the default credentials: [**_username: admin, password: admin] _**
+
+![alt_text](images/image2.png "image_tooltip")
+
+### **3.1 Configure Hadoop**
+
+Navigate to **Services** -> **HDFS** -> **CONFIGS** -> **ADVANCED **as shown below
+
+![alt_text](images/image3.png "image_tooltip")
+
+Navigate to **Custom core-site** to configure MinIO parameters for `_s3a_` connector
+
+![alt_text](images/image4.png "image_tooltip")
+
+Add the following optimal entries for _core-site.xml_ to configure _s3a_ with **MinIO**. Most important options here are
+
+*   _fs.s3a.access.key=minio_ (Access Key to access MinIO instance, this is obtained after the deployment on k8s)
+*   _fs.s3a.secret.key=minio123_ (Secret Key to access MinIO instance, this is obtained after the deployment on k8s)
+*   _fs.s3a.endpoint=http://minio-address/_
+*   _fs.s3a.path.style.acces=true_
+
+Rest of the other optimization options are discussed here
+
+*   [https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/index.html](https://hadoop.apache.org/docs/current/hadoop-aws/tools/hadoop-aws/index.html)
+*   [https://hadoop.apache.org/docs/r3.1.1/hadoop-aws/tools/hadoop-aws/committers.html](https://hadoop.apache.org/docs/r3.1.1/hadoop-aws/tools/hadoop-aws/committers.html)
+
+![alt_text](images/image5.png "image_tooltip")
+
+Once the config changes are applied, proceed to restart **Hadoop** services.
+
+![alt_text](images/image6.png "image_tooltip")
+
+### **3.2 Configure Spark2**
+
+Navigate to **Services** -> **Spark2** -> **CONFIGS** as shown below
+
+![alt_text](images/image7.png "image_tooltip")
+
+Navigate to “**Custom spark-defaults**” to configure MinIO parameters for `_s3a_` connector
+
+![alt_text](images/image8.png "image_tooltip")
+
+
+Add the following optimal entries for _spark-defaults.conf_ to configure Spark with **MinIO**.
+
```

Comment:
Same feedback as above for adding an additional code block to allow for copy/paste